e2e/generate_prompt.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { test, expect, Page } from '@playwright/test';
+import dotenv from 'dotenv';
+
+// Load env from backend for REGISTER_KEY
+dotenv.config({ path: 'backend/.env' });
+
+const output_locator = 'textarea[placeholder*="AI generated content"]';
+const theme_input = 'Sci-fi';
+const category_input = 'category input from playwright';
+const prompt_input = 'prompt input from playwright';
+
+const uuid = () => Math.random().toString(36).substring(2, 12);
+const username = uuid();
+
+let registered = false;
+test.beforeEach(async ({ page }) => {
+    // Register once per worker
+    if (!registered) {
```

Comment:
I must notify that now you have that top level if else instead of using beforeAll and beforeEach. I don't even know if Playwright supports beforeAll/before but if it does you could consider refactoring this.

Reply:
Playwright before (analogue of beforeAll of Cypress) does not support a page parameter, as it is generic over tests (unlike Cypress where everything is in one context), so you would have to manually create a new browser context. It's doable but requires some effort.